app/src/main/java/org/fossasia/openevent/app/data/EventRepository.java

```diff
@@ -63,27 +69,74 @@ public EventRepository(IUtilModel utilModel, ICacheModel cacheModel, EventServic
             .observeOn(AndroidSchedulers.mainThread());
     }
 
+    private <T extends BaseModel> Observable<T> getAbstractObservable(boolean reload, Observable<T> diskObservable, Observable<T> networkObservable) {
+        return
+            Observable.defer(() -> {
+                if (reload)
+                    return Observable.empty();
+                else
+                    return diskObservable
+                        .doOnNext(item -> Timber.d("Loaded %s From Disk", item.getClass()));
+            }).switchIfEmpty(
+                Observable.just(utilModel.isConnected())
+                    .flatMap(connected -> {
+                        if (connected)
+                            return networkObservable
+                                .doOnNext(item -> Timber.d("Loaded %s From Network", item.getClass()));
+                        else
+                            return Observable.error(new Throwable(Constants.NO_NETWORK));
+                    }))
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread());
+    }
+
     @Override
     public Observable<User> getOrganiser(boolean reload) {
         return getData(() -> eventService.getUser(authorization), ORGANIZER, reload);
     }
 
     @Override
     public Observable<Event> getEvent(long eventId, boolean reload) {
-        return getData(() -> eventService.getEvent(eventId), EVENT + eventId, reload);
+        Observable<Event> diskObservable = Observable.defer(() ->
+            databaseRepository
+                .getItem(Event.class, Event_Table.id.eq(eventId))
+        );
+
+        Observable<Event> networkObservable = Observable.defer(() ->
+            eventService
+                .getEvent(eventId)
+                .doOnNext(event ->
+                    databaseRepository
+                        .save(event)
+                        .subscribe()
+                )
+        );
+
+        return getAbstractObservable(reload, diskObservable, networkObservable);
     }
 
     @Override
-    public Observable<List<Attendee>> getAttendees(long eventId, boolean reload) {
-        return getData(() -> eventService.getAttendees(eventId, authorization), ATTENDEES + eventId, reload);
+    public Observable<Event> getEvents(boolean reload) {
+        Observable<Event> diskObservable = Observable.defer(() ->
+            databaseRepository.getAllItems(Event.class)
+        );
+
+        Observable<Event> networkObservable = Observable.defer(() ->
+            eventService.getEvents(authorization)
+                    .flatMapIterable(events -> events)
+                    .doOnNext(event -> {
+                        event.setComplete(false);
+                        databaseRepository.save(event).subscribe();
```

Comment:
Please see corresponding tests for updating on reload. That's what tests are for. Besides `databaseRepository.save(event).subscribe()` should give a clue